contracts/Aavegotchi/facets/ERC721BuyOrderFacet.sol

```diff
@@ -0,0 +1,216 @@
+        );
+        require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+        if (erc721BuyOrder.duration > 0) {
+            require(erc721BuyOrder.timeCreated + erc721BuyOrder.duration >= block.timestamp, "ERC721BuyOrder: Already expired");
+        }
+
+        LibBuyOrder.cancelERC721BuyOrder(_buyOrderId);
+    }
+
+    function executeERC721BuyOrder(uint256 _buyOrderId, address _erc721TokenAddress, uint256 _erc721TokenId, uint256 _priceInWei) external {
+        address sender = LibMeta.msgSender();
+        ERC721BuyOrder storage erc721BuyOrder = s.erc721BuyOrders[_buyOrderId];
+
+        require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+        require(erc721BuyOrder.erc721TokenAddress == _erc721TokenAddress, "ERC721BuyOrder: ERC721 token address not matched");
+        require(erc721BuyOrder.erc721TokenId == _erc721TokenId, "ERC721BuyOrder: ERC721 token id not matched");
+        require(erc721BuyOrder.priceInWei == _priceInWei, "ERC721BuyOrder: Price not matched");
+        require(sender == s.aavegotchis[erc721BuyOrder.erc721TokenId].owner, "ERC721BuyOrder: Only aavegotchi owner can call this function");
+        require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+        if (erc721BuyOrder.duration > 0) {
+            require(erc721BuyOrder.timeCreated + erc721BuyOrder.duration >= block.timestamp, "ERC721BuyOrder: Already expired");
+        }
+
+        // disable for gotchi in lending
+        uint256 category = LibSharedMarketplace.getERC721Category(erc721BuyOrder.erc721TokenAddress, erc721BuyOrder.erc721TokenId);
+        if (category == LibAavegotchi.STATUS_AAVEGOTCHI) {
+            require(
+                !LibGotchiLending.isAavegotchiLent(uint32(erc721BuyOrder.erc721TokenId)),
+                "ERC721BuyOrder: Not supported for aavegotchi in lending"
+            );
+        }
+
+        // hash validation
+        require(
+            erc721BuyOrder.validationHash ==
+                LibBuyOrder.generateValidationHash(erc721BuyOrder.erc721TokenAddress, erc721BuyOrder.erc721TokenId, erc721BuyOrder.validationOptions),
+            "ERC721BuyOrder: Invalid buy order"
+        );
+
+        BaazaarSplit memory split = LibSharedMarketplace.getBaazaarSplit(erc721BuyOrder.priceInWei, new uint256[](0), [10000, 0]);
+
+        LibSharedMarketplace.transferSales(
+            SplitAddresses({
+                ghstContract: s.ghstContract,
+                buyer: address(this),
+                seller: sender,
+                affiliate: address(0),
+                royalties: new address[](0),
+                daoTreasury: s.daoTreasury,
+                pixelCraft: s.pixelCraft,
+                rarityFarming: s.rarityFarming
+            }),
+            split
+        );
+
+        s.aavegotchis[erc721BuyOrder.erc721TokenId].locked = false;
```

Comment:
This should be nested within the if statement below. If a buy order is executed for a non-Aavegotchi NFT, we should not be updating the locked state of the Aavegotchi with that tokenID. @orionstardust